### PR TITLE
Allow notifications to use the alarm stream

### PR DESCRIPTION
**Pull request by dshokouhi**
Fixes: #561 

The best way to make sure a notification stays as expected while also respecting Android 8+ channels lets use a specific channel to bypass the devices ringer mode.  The notification will make a sound (regardless if the device is on vibrate/silent) and behave much like an alarm, the stream will stop once the user interacts with their device. The default sound will be the default alarm which may be enough for most users.  There is also code to support legacy devices too, we are just reading the channel name and changing the stream accordingly.

Example:

```
message: Time to wake up!
data:
  channel: alarm_stream
```

Docs: https://github.com/home-assistant/companion.home-assistant/pull/354

### CODE REVIEW

**Comment by dshokouhi** [comment]
Thought of another interesting use case for this.....users can use HA to create their own alarms on the device entirely controlled from HA :)  Don't even need to touch an alarm app, just set it in HA and have the house wake you up as needed.

File: app/src/full/java/io/homeassistant/companion/android/notifications/MessagingService.kt [hunk 1]
```diff
@@ -206,8 +209,13 @@ class MessagingService : FirebaseMessagingService() {
...
         val notificationBuilder = NotificationCompat.Builder(this, channelId)
             .setSmallIcon(R.drawable.ic_stat_ic_notification)
-            .setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION))
 
+        if (data["channel"] == "alarm_stream") {
+            notificationBuilder.setCategory(Notification.CATEGORY_ALARM)
+            notificationBuilder.setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM), AudioManager.STREAM_ALARM)
+        } else {
+            notificationBuilder.setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION))
+        }
```
Comment (JBassett) [inline app/src/full/java/io/homeassistant/companion/android/notifications/MessagingService.kt:218]: We should break the sound handling into a method like the rest of the items.  I think the category can go in there for now as well.

**Review by JBassett — CHANGES_REQUESTED** [review]

**Review by JBassett — APPROVED** [review]

**Comment by cwalger** [comment]
Hello, this looked promising but I am using a Samsung S10e and the latest version of the app from the play store and the notifications do not play a sound when the phone is set to vibrate or to mute.  When my phone is set to sound, the notifications do use the alarm ringtone, so it is partially working.  Thank you

**Comment by dshokouhi** [comment]
@cwalger please do not use merged PRs to discuss issues, there is already an open issue for this that you can read through and provide feedback and logs as requested: https://github.com/home-assistant/android/issues/1186

**Comment by cwalger** [comment]
Sorry, noted, and thank you for the link.  I read through a few threads elsewhere that pointed to this as the solution.

**Pull request merged**